### PR TITLE
feat: add support separately included authors

### PR DESCRIPTION
**Pull request by vandie**
Looking at the indie web wiki, it's possible to include author information [on a separate page](https://indieweb.org/authorship#Dedicated_author_page). This seems like a reasonable thing to support and as such this PR adds support for it.

### CODE REVIEW

File: src/classes/web-mention-handler.class.ts [hunk 1]
```diff
@@ -97,6 +98,13 @@ export class WebMentionHandler implements IWebMentionHandler{
     // early as we have already deleted any stored mentions with this target and source
     if(!mentionedUrls) return null;
 
+    // Not every post has an author, some pages also have authors on a
+    // seperate page, as such we should normalize it a local author if possible
+    await Promise.all(hEntries.map(async (entry, index) => {
+      if(!entry.author) return;
+      hEntries[index].author = await normalizeAuthors(entry);
```
Comment (BeardScript) [inline src/classes/web-mention-handler.class.ts:105]: Isn't this the same as entry. author = ...?

File: src/functions/get-external-author.function.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { getExternalAuthor } from "./get-external-author.function";
+let fetchMock:jest.SpyInstance;
+describe('getExternalAuthor', () => {
+  beforeEach(() => {
+    fetchMock = jest.spyOn(require('./fetch-html.function'), 'fetchHtml').mockImplementation((url) => {
+      if(url === 'https://example.org') return {status: 200, html: `<html><body><div class="h-card"><img class="u-photo" src="https://example.org/images/mike.png" alt="Michael Walter Van Der Velden"><h1><a href="/" class="u-url p-name">Michael Walter Van Der Velden</a></h1></div></body></html>`};
```
Comment (BeardScript) [inline src/functions/get-external-author.function.test.ts]: I would add proper html formatting here

**Review by BeardScript — COMMENTED** [review]
Just some nitpicks looks good to go!

File: src/classes/web-mention-handler.class.ts (hunk 1 above)
Comment (vandie) [reply src/classes/web-mention-handler.class.ts:105]: doing it this way helps avoid creating an index on the object if there isn't an author.
don't want a load of objects with
```js
{ author: undefined }
```
in them

**Pull request merged**